network/common/src/main/java/org/apache/spark/network/sasl/SaslClientBootstrap.java

```diff
@@ -62,13 +76,26 @@ public void doBootstrap(TransportClient client) {
         byte[] response = client.sendRpcSync(buf.array(), conf.saslRTTimeoutMs());
         payload = saslClient.response(response);
       }
+
+      if (encrypt) {
+        if (!SparkSaslServer.QOP_AUTH_CONF.equals(saslClient.getNegotiatedProperty(Sasl.QOP))) {
+          throw new RuntimeException(
+            new SaslException("Encryption requests by negotiated non-encrypted connection."));
+        }
+        SaslEncryption.addToChannel(channel, saslClient, conf.maxSaslEncryptedBlockSize());
+        saslClient = null;
+        logger.debug("Channel {} configured for SASL encryption.", client);
+      }
     } finally {
-      try {
-        // Once authentication is complete, the server will trust all remaining communication.
-        saslClient.dispose();
-      } catch (RuntimeException e) {
-        logger.error("Error while disposing SASL client", e);
+      if (saslClient != null) {
```

Comment:
rather than setting `saslClient = null` in the `encrypt` block, and then checking for null here, why not just use `if (!encrypt)` here?  If there is not another reason to `saslClient = null`, I think that would be clearer.